### ISSUE TITLE
Feature Request: fs::remove_all

### ISSUE THREAD

**Issue by Nokel81** [async-rs/async-std#692]
Would having the following function be a good addition to this library?

```rust
/// This function will either remove the file specified at `path` or remove the
/// directory and all of its contents specified at `path`
pub async fn remove_all<P: AsRef<Path>>(path: P) -> io::Result<()> { ... }
```

**Comment by skade** [comment]
In general, async-std tries to track rusts libstd as close as possible. libstd doesn't provide `remove_all`.

Is there any particular reasons this needs to be in the library and cannot be implemented outside of it?

**Comment by Nokel81** [comment]
I think that this should be in the library and not outside of it because it fits very much into the other `remove_*` functions. Though I can definitely understand not wanting to add it if the `std` doesn't have it. I wonder if they might add it.... (I'll ask).

**Comment by skade** [comment]
@Nokel81 I agree that it is a reasonable addition to libstd. I'll leave this issue open for reference.
